docs/docs/guides/evaluation/scorers.md

```diff
@@ -0,0 +1,615 @@
+# Evaluation Metrics
+
+## Evaluations in Weave
+In Weave, Scorers are used to evaluate AI outputs and return evaluation metrics. They take the AI's output, analyze it, and return a dictionary of results. Scorers can use your input data as reference if needed and can also output extra information, such as explanations or reasonings from the evaluation.
+
+Scorers are passed to a `weave.Evaluation` object during evaluation. There are two types of Scorers in weave:
+
+1. **Function-based Scorers:** Simple Python functions decorated with `@weave.op`.
+2. **Class-based Scorers:** Python classes that inherit from `weave.Scorer` for more complex evaluations.
+
+Scorers must return a dictionary and can return multiple metrics, nested metrics and non-numeric values such as text returned from a LLM-evaluator about its reasoning.
+
+## Create your own Scorers
+### Function-based Scorers
+These are functions decorated with `@weave.op` that return a dictionary. They're great for simple evaluations like:
+
+```python
+@weave.op
+def evaluate_uppercase(text: str):
+    return {"text_is_uppercase": text.isupper()}
+
+my_eval = weave.Evaluations(..., scorers=[evaluate_uppercase])
```

Comment:
same here, let's just flesh out the example with something copy-able.

We have the `#highlight-next-line` pragma to highlight the weave-specific bits